tests/integration/test_issue_1726.py

```diff
@@ -0,0 +1,68 @@
+# Copyright 2022 Pants project contributors (see CONTRIBUTORS.md).
+# Licensed under the Apache License, Version 2.0 (see LICENSE).
+
+from __future__ import absolute_import
+
+import os
+import subprocess
+import sys
+from textwrap import dedent
+
+import pytest
+
+from pex.common import safe_open
+from pex.testing import run_pex_command
+from pex.typing import TYPE_CHECKING
+
+if TYPE_CHECKING:
+    from typing import Any
+
+
+@pytest.mark.skipif(
+    sys.version_info[:2] < (3, 7),
+    reason="The jaraco-collections 3.5.1 distribution requires Python >=3.7",
+)
+def test_check_install_issue_1726(tmpdir):
+    # type: (Any) -> None
+
+    src = os.path.join(str(tmpdir), "src")
+    with safe_open(os.path.join(src, "setup.py"), "w") as fp:
+        fp.write(
+            dedent(
+                """\
+                from setuptools import setup
+
+                setup(
+                    name="pex-test",
+                    version='0.1',
+                    install_requires=[
+                        "jaraco-collections==3.5.1",
```

Comment:
The error only reproed for me when the project name came via setuptools metadata. In other words if you directly specify this requirement or the full pinned requirements in the [example repo ](https://github.com/Djailla/pex_bug) @Djailla was so kind as to provide, on the Pex command line, everything works. This is beaucoup ironic since jaraco is the primary setuptools author!